### ISSUE TITLE
Can't get multiple transactions in one block when using -b

### ISSUE THREAD

**Issue by jpmonettas** [trufflesuite/ganache-cli#707]
Running ganache-cli(6.1.8) with -b 5 and submitting 4 transactions in a row should result in the 4 transactions being mined in the same block. Instead I'm seeing each transaction being mined in a separate block.

* I asked about this on [gitter](https://gitter.im/ConsenSys/truffle): [X]
* Link to my question on gitter: [January 7, 2020 1:37 PM](https://gitter.im/ConsenSys/truffle?at=5e14b3ac04adee486b49491e)

> hello everybody!
> I'm running ganache-cli (6.1.8) with -b 5 and doing a bunch of eth_sendTransaction together. I see in ganache console that the transactions are being sent one after the other very quickly but I'm also seeing ganache processing each of them when mining a different block, so all transactions end up in different blocks which is not what I'm trying to accomplish.
> Any ideas?

## Expected Behavior

Each transaction should be mined in the same block when ganache-cli runs with -b t and the sum of all times between transaction submission is < t.

## Current Behavior

Each transaction is mined in a separate block.

## Steps to Reproduce (for bugs)

1. Run ganache-cli like :
```
ganache-cli -b 5 -v | grep et"eth_sendTransaction\|Transaction\|Block Number"
```

so you can see transaction submissions and transactions being mined with block number.

2. Submit 4 transactions in a row using web3js 
3. Check in ganache log that all transactions were sent one after the other  but it mines one block per transaction. It mines one block every 5 seconds (this is expected) but it includes 1 tx per block. 

## Your Environment
* Version used: 6.1.8
* Version of Truffle/Remix/Other tools used:
* NodeJS Version: v11.14.0
* Operating System and version (include distro if Linux): Debian Linux

**Comment by nicholasjpaterno** [comment]
@jpmonettas Thanks for opening this issue.  I was unable to reproduce this, is there a chance you can link to a repo with a minimum viable test?  My hunch is that you are sending the transaction and awaiting the receipt which will only be available after the tx was included in a block.

**Comment by jpmonettas** [comment]
> My hunch is that you are sending the transaction and awaiting the receipt which will only be available after the tx was included in a block

@nicholasjpaterno  how can I see all 4 eth_sendTransaction in ganache console output together immediately and after 5 seconds it start mining blocks putting each of them in separate blocks?

I'm seeing this in the console when running ganache-cli in verbose mode.

**Comment by nicholasjpaterno** [comment]
@jpmonettas have a look at this case in our test suite: https://github.com/trufflesuite/ganache-core/blob/develop/test/mining.js#L154-L184

**Comment by jpmonettas** [comment]
@nicholasjpaterno I created a minimal example and wasn't able to reproduce. I'm still not sure of what is causing it but will keep troubleshooting, so closing this for now! 


Thanks for your time!

**Comment by jpmonettas** [comment]
I was able to reproduce the issue. Here is a minimal example repo : https://github.com/jpmonettas/ganache-test

The issue appears when you explicitly add a gas limit  when sending the transaction. Without the gas option it works fine.

**Comment by jpmonettas** [comment]
Oh I think I'm hitting this exact scenario https://github.com/trufflesuite/ganache-core/blob/develop/test/mining.js#L186

But shouldn't ganache be using tx gas usage instead of tx gas limit to create a new block because it doesn't fit in current?

**Comment by nicholasjpaterno** [comment]
@jpmonettas You're absolutely right!  I'm working on a fix for this now.  In the meantime you can use an `eth_estimateGas` as a workaround to get an accurate `gasLimit` for your transactions - this should enable you to test several transactions within your blocks!
